wire-ios-sync-engine/Source/Synchronization/EventProcessor.swift

```diff
@@ -225,3 +261,32 @@ extension Notification.Name {
     /// Published after the last event has been processed.
     static let eventProcessorDidFinishProcessingEventsNotification = Self("EventProcessorDidFinishProcessingEvents")
 }
+
+private actor ProcessedEventList {
+
+    private var hashes = Set<Int64>()
+
+    // A full list would contain approx 80kB.
+    private let capacity = 10_000
+
+    func addEvent(_ event: ZMUpdateEvent) {
+        guard let hash = event.contentHash else {
+            return
+        }
+
+        if hashes.count >= capacity, let randomElement = hashes.randomElement() {
+            hashes.remove(randomElement)
+        }
+
+        hashes.insert(hash)
+    }
+
+    func containsEvent(_ event: ZMUpdateEvent) -> Bool {
+        guard let hash = event.contentHash else {
+            return false
```

Comment:
Same as above. I would assert and log error here.

Reply:
done 75842748fdfcc876752f5fa3183f1cd7c8bc095d